From 46e1a3caff42f1118e2dc31618366d14beb069fb Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 7 Dec 2022 01:52:45 +0100
Subject: [PATCH 7/7] RunGrepTest: work around MSYS2 idiosyncrasy

When calling the MSYS2 version of `echo` from a non-MSYS program, the
command-line is parsed in a non-standard manner. In particular when
spawning `echo` with a single argument that contains a newline, that
argument is parsed as if it were a command-line and the argument is
split into two.

Let's avoid that by using `echo -e ...\n` instead.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 RunGrepTest | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/RunGrepTest b/RunGrepTest
index 20ad33a5..f32d72c9 100755
--- a/RunGrepTest
+++ b/RunGrepTest
@@ -1071,7 +1071,7 @@ if $valgrind $vjs $pcre2grep --help | $valgrind $vjs $pcre2grep -q 'callout scri
   $valgrind $vjs $pcre2grep '(T)(?C"|$0:$1$n")' $srcdir/testdata/grepinputv >>testtrygrep
   echo "RC=$?" >>testtrygrep
   echo "--- Test 4 ---" >>testtrygrep
-  $valgrind $vjs $pcre2grep '(T)(?C"echo|$0:$1$n")' $srcdir/testdata/grepinputv >>testtrygrep
+  $valgrind $vjs $pcre2grep '(T)(?C"echo|-e|$0:$1\n")' $srcdir/testdata/grepinputv >>testtrygrep
   echo "RC=$?" >>testtrygrep
   echo "--- Test 5 ---" >>testtrygrep
   $valgrind $vjs $pcre2grep '(T)(?C"|$1$n")(*F)' $srcdir/testdata/grepinputv >>testtrygrep
@@ -1097,7 +1097,7 @@ if $valgrind $vjs $pcre2grep --help | $valgrind $vjs $pcre2grep -q 'callout scri
     $valgrind $vjs $pcre2grep -u '(T)(?C"|$0:$x{a6}$n")' $srcdir/testdata/grepinputv >>testtrygrep
     echo "RC=$?" >>testtrygrep
     echo "--- Test 2 ---" >>testtrygrep
-    $valgrind $vjs $pcre2grep -u '(T)(?C"echo|$0:$x{a6}$n")' $srcdir/testdata/grepinputv >>testtrygrep
+    $valgrind $vjs $pcre2grep -u '(T)(?C"echo|-e|$0:$x{a6}\n")' $srcdir/testdata/grepinputv >>testtrygrep
     echo "RC=$?" >>testtrygrep
 
     if [ $nonfork = 1 ] ; then
-- 
2.48.1.windows.1

